bin/downgrade

```diff
@@ -290,6 +291,8 @@ process_term() {
     } >&2
   elif (("${#candidates[@]}" == 1)); then
     choice=${candidates[0]}
+  elif ((DOWNGRADE_TO_ANY)); then
+    choice=${candidates[0]}
```

Comment:
I worry about the word "any" and description "just pick any candidate", since that doesn't indicate how the choice is made. It's like "oh, just pick _any_ candidate, I don't care which". I forget which way the results are ordered, but whichever `[0]` is (newest or oldest), I would indicate that in the option name and description somehow.

Bonus points for supporting both via (e.g.) `--select=oldest|newest`:

```bash
elif [[ -n "$DOWNGRADE_SELECT" ]]; then
  case "$DOWNGRADE_SELECT" in
    oldest) choice=${candidates[0]} ;;
    newest) choice=${candidates[-1]} ;;
    *) # error
    ;;
  esac
```

Reply:
Good point. I have made sure it will pick a version from the cache if there is one. Right now it does pick the oldest which is probably not what people want. In my use case I did specify precise versions so I didn't care initially:

https://github.com/archlinux-downgrade/downgrade/issues/118#issuecomment-1746830312

I will change it to picking the newest matching version for now but then this would still be the newest locally available version (if applicable) and the newest ala version otherwise

Reply:
The readme now documents the behaviour like this

> Never prompt for version or location, pick automatically instead. Allows for
> noninteractive downgrades when used in conjunction with --ignore
> If a package matching the version filter is found in the cache, pick the
> latest matching version from cache. Íf no matches are found in cache or
> --ala-only was specified, then pick the latest matching version from ala.

While `--help` and completion just call it "automatic package selection"

We can rename the option to `--autoselect`. I would refrain from --select=oldest|newest because if the version is not specifically mentioned, newer is better and I think the behaviour of picking from the cache before ala makes a lot of sense for when both are enabled.